### PR TITLE
FSDP tutorial

### PR DESCRIPTION
**Pull request by HamidShojanazeri**
PR for  Getting started with FSDP tutorial.

### CODE REVIEW

File: intermediate_source/FSDP_tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+It can be observed that the peak memory usage on each device is smaller compared to FSDP without auto wrap policy applied, from ~75 MB to 66 MB.
+
+.. figure:: /_static/img/distributed/FSDP_autowrap.gif
+   :width: 100%
+   :align: center
+   :alt: FSDP peak memory
+
+   FSDP Peak Memory Usage using Auto_wrap policy
+
+Compared with FSDP without setting the auto_wrap_ploicy where all the layers were placed in one FSDP unit, we can see the impact of communication overhead in CUDA event timing, as it took a bit longer in this case.
```
Comment (zhaojuanmao) [inline intermediate_source/FSDP_tutorial.rst:379]: we can remove this, as the last section explained the timing in high level?

File: intermediate_source/FSDP_tutorial.rst [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+
+
+Compare it with DDP, if in 2.4 we just normally wrap the model in ddp, saving the changes in “DDP_mnist.py”.
+
+.. code-block:: python
+    model = Net().to(rank)
+    model = DDP(model)
+
+Running the DDP, the CUDA events have taken slightly less time than the FSDP due to less communication overhead. But this is the result for small models on slow network clusters. 
+For large model training on fast network clusters with FSDP internal optimizations, the communication overhead could be mitigated significantly.
```
Comment (zhaojuanmao) [inline intermediate_source/FSDP_tutorial.rst:405]: we can remove this, as the last section explained the timing in high level?

**Review by zhaojuanmao — COMMENTED** [review]
looks overall great! thanks for adding this! just some minor comments

**Comment by HamidShojanazeri** [comment]
Thanks @zhaojuanmao, comments are addressed.

**Review by zhaojuanmao — APPROVED** [review]
awesome! let's land it!

**Comment by mrshenli** [comment]
Hey @brianjo, shall we land this PR? Thanks!

**Pull request merged**